### ISSUE TITLE
[Bug]: Array data

### ISSUE THREAD

**Issue by c0nsoleg** [airjp73/remix-validated-form#116]
### Which packages are impacted?

- [X] `remix-validated-form`
- [ ] `@remix-validated-form/with-zod`
- [ ] `@remix-validated-form/with-yup`
- [ ] `zod-form-data`

### What version of these packages are you using?

    "remix-validated-form": "^4.4.3",
    "@remix-validated-form/with-zod": "^2.0.1",

### Please provide a link to a minimal reproduction of the issue.

https://www.remix-validated-form.io/arrays-and-nested

### Steps to Reproduce the Bug or Issue

Go to example, add new todo item and remove te first one.

### Expected behavior

The first element (with filled inputs) should be removed and only visible should be the new added, empty one, but it do otherwise (or more likely it removes first one but only from render and data still stand there).

### Platform

- OS: Windows
- Browser: Firefox
- Version: 100.0.1

**Comment by airjp73** [comment]
Good catch! I'll look into this.

**Comment by c0nsoleg** [comment]
There is also a weird behaviour when you add new one and type only note field and then remove the first one it is going to join notes to the first one leaving its title

**Comment by airjp73** [comment]
This appears to be related to the default values. If an input doesn't have a default value and then you add one, it will update the input to have that value. The first item in the list has a default value, but any items you add after don't. If you delete the first item, the `name` of the inputs in the second item will change, causing them to gain a default value. So the second items values are updated to be the default values of the first item.

I'll need to spend some time to consider solutions for this.

**Comment by airjp73** [comment]
For now, I've removed the example from the site. I'm working on a `FieldArray` helper that I'm hoping will make this use-case fully possible.

**Comment by airjp73** [comment]
It'll probably be a couple more weeks. I'm in the process of moving, so I haven't had much time for non-work coding.

**Comment by airjp73** [comment]
I still have documentation to write for the new feature, but I've released a beta with the fieldarray feature (4.6.0-beta.0) if anyone wants to try it.

**Comment by keithrowe** [comment]
I'm using this successfully with 4.6.0-beta.0.

However, in VSCode, I'm getting an error:
```
'FieldArray' cannot be used as a JSX component.
Its return type 'ReactNode' is not a valid JSX element.
Type 'undefined' is not assignable to type 'Element | null'
```

It seems to run fine, but I don't know how to clear the error.

I got the same error when I copied https://github.com/airjp73/remix-validated-form/tree/fieldarray/apps/test-app/app/routes/field-array.tsx into my project.

**Comment by zelid** [comment]
Confirm TS error:

```
'FieldArray' cannot be used as a JSX component.
Its return type 'ReactNode' is not a valid JSX element.
Type 'undefined' is not assignable to type 'Element | null'
```
Temporary solution to ignore TS error:
```
{/* @ts-ignore */}
<FieldArray name="variations">
```

**Comment by airjp73** [comment]
Released (finally) in 4.6.0 🎉
